🚧Create outside contributor docs (#409)
KinsonDigital/Velaptor: CONTRIBUTING.md

- **DON'T** commit code you didn't write without following its license and following our guidelines for doing so in the Contributor License Agreement. If you are unable to license some code, don't commit it.
- **DON'T** surprise us with big pull requests or big API changes without talking to us first!
- **DON'T** make PRs for legal or administrative documents, such as the license, file headers, or code of conduct. If you see something off, let us know and we will look into changing it.

<h2 style="border:0;font-weight:bold" align="center">Breaking Changes</h2>

Contributions must maintain API signature and behavioral compatibility. Contributions that include breaking changes will be rejected. There are exceptions to this, such as if the contribution is made against a major/breaking version branch (such as 2.0 instead of master), however, this is subject to the team's approval. At risk of sounding like a broken record, talk to us about your idea first!

<h2 style="border:0;font-weight:bold" align="center">Branching</h2>

> Go [here](./Branching.md) for more information on <span style="color: #66B2FF;font-weight:bold">feature</span> and <span style="color: #9E269E;font-weight:bold">preview feature</span> branches.

Reviewer: The "here" link does not work.

<h2 style="border:0;font-weight:bold" align="center">Creating Early Pull Request</h2>

Refer to the **DOs and DON'Ts** section for certain details about pull requests.

**Velaptor** encourages and uses [Early Pull Requests](https://medium.com/practical-blend/pull-request-first-f6bb667a9b6). Please don't wait until you're finished with your work before creating a PR!

1. If you have not already, do step 1 in the **Setting Things Up** section above.
2. If you have not already, do step 2 in the **Setting Things Up** section above.
3. If you have not already, do step 3 in the **Setting Things Up** section above.
4. If you have not already, do step 1 in the **Create a Branch to Contribute** section above.
5. If you have not already, do step 2 in the **Create a Branch to Contribute** section above.
6. Make your changes.
7. Once you are done with your changes, open a [**draft pull request**](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests). Do this **BEFORE** you start working.

>    - Please make sure that the title of the pull request matches the title of the associated issue _**EXACTLY.**_

Reviewer: Sentence structure
```suggestion
   - Please make sure that the title of the pull request _**EXACTLY.**_ matches the title of the associated issue. 
```

   - Use one of the PR templates below depending on if you are doing a regular feature or a preview feature.
     - [Feature PR Template](https://raw.githubusercontent.com/KinsonDigital/.github/master/.github/PULL_REQUEST_TEMPLATE/feature-pr-template.md)
     - [Preview Feature PR Template](https://raw.githubusercontent.com/KinsonDigital/.github/master/.github/PULL_REQUEST_TEMPLATE/preview-feature-pr-template.md)
8. Make your commits in small, incremental steps with clear descriptions.
9. All unit tests must pass before a PR will be completed.
10. Make sure that the code follows the coding standards.
   * Pay attention to the warnings in **Visual Studio** and/or **JetBrains Rider**!!

>    * Refer to the *.editorconfig* files in the code base for rules.

Reviewer: The .editorconfig is not a link.

11. When you are finished with your changes, tag a maintainer in the PR comments and ask for a review!

<h2 style="border:0;font-weight:bold" align="center">"Help wanted" & "Good first issue"</h2>

> If the core team is unable to champion an issue, they will mark it with a "help wanted" label. This indicates that any external contributor may pick up on this issue and implement it as part of a contribution. However, some "help wanted" issues may require intermediate knowledge of the codebase, area, and/or ecosystem; or may have uncertainty surrounding implementation details (if this is the case talk to us in [Discord](https://discord.gg/qewu6fNgv7) or in a discussion issue). Because of this, we also mark some issues with the label "good first issue" which indicates that an issue is straightforward, and is a good place to start if you're interested in contributing but new to the codebase.

Reviewer: Reword

```suggestion
If the core team is unable to champion an issue, they will mark it with a "help wanted" label. This indicates that any external contributor may pick it up and implement it as part of a contribution. However, some "help wanted" issues may require intermediate knowledge of the codebase, area, and/or ecosystem; or may have uncertainty surrounding implementation details.  If this is the case, talk to us in [Discord](https://discord.gg/qewu6fNgv7) or in a discussion issue. We also mark some issues with the label "good first issue" which indicates that an issue is straightforward and is a good place to start if you're interested in contributing but are new to the codebase.
```

<h3 style="border:0;font-weight:bold" align="left">Resources</h3>

- [Forking a Repository](https://docs.github.com/en/get-started/quickstart/fork-a-repo)
- [Working With Forks](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/about-forks)
- [Early Pull Requests](https://medium.com/practical-blend/pull-request-first-f6bb667a9b6)
- [Discord Server](https://discord.gg/qewu6fNgv7)

> - [Main Branching Docs](./Branching.md)

Reviewer: This link is broken

- **DON'T** start working on a pull request until you get confirmation from a project maintainer and the issue is assigned to you.  You do not want to end up doing a bunch of work for nothing just to find out that you are not allowed to work on the issue.
- **DON'T** make PRs that don't directly affect the end user, such as style changes. These are best done as part of a PR related to the area in question. Documentation is fine (and encouraged!), as this is useful to the end user.
- **DON'T** commit code you didn't write without following its own license and without following our guidelines in the Contributor License Agreement. If you are unable to license code, don't commit it.
- **DON'T** surprise us with big pull requests or big API changes without talking to us first!
- **DON'T** make PRs for legal or administrative documents, such as the license, file headers, or code of conduct. If something is off, let us know and we will look into changing it.

<h2 style="border:0;font-weight:bold" align="center">Breaking Changes</h2>

> Contributions must maintain API signature and behavioral compatibility. Contributions that include breaking changes will be rejected, however, there are exceptions. For example, if the contribution is made against a major/breaking version branch (such as 2.0 instead of master).  Breaking changes are subject to the team's approval. At risk of sounding like a broken record, talk to us about your idea first!

Reviewer: ```suggestion
Contributions must maintain API signature and behavioral compatibility. Contributions that include breaking changes will be rejected, however, there are exceptions. One example would be if the contribution is made against a major/breaking version branch (such as 2.0 instead of master).  Breaking changes are subject to the team's approval. At risk of sounding like a broken record, talk to us about your idea first!
```
